tsdb/engine/tsm1/engine_test.go

```diff
@@ -816,6 +816,25 @@ func TestEngine_CreateCursor_Descending(t *testing.T) {
 	}
 }
 
+func makeBlockTypeSlice(n int) []byte {
+	r := make([]byte, n)
+	b := tsm1.BlockFloat64
+	m := tsm1.BlockUnsigned + 1
+	for i := 0; i < len(r); i++ {
+		r[i] = b % m
+	}
+	return r
+}
+
+func BenchmarkBlockTypeToInfluxQLDataType(b *testing.B) {
+	t := makeBlockTypeSlice(100)
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < len(t); j++ {
+			tsm1.BlockTypeToInfluxQLDataType(t[j])
```

Comment:
Are you sure this can't get optimised away by the compiler? That used to be a problem so I always got into the habit of defining a `var result influxql.Unknown` somewhere outside the benchmark.

Reply:
Good catch – if the function was not inlined, it would have made no difference. We can deduce the inner loop overhead is thus about 45ns :)

Original benchmark was about 212ns, less loop overhead is 167ns. I added an assignment and it jumped from 45ns to 70ns. So about 30ns vs 167ns or about a 5.5x improvement.

Reply:
Still a good improvement.